.github/workflows/ci.yml

```diff
@@ -30,7 +30,8 @@ env:
   BUILDTOOLS_VERSION: '3.0.0'
   BUILDIFIER_SHA256SUM: 'e92a6793c7134c5431c58fbc34700664f101e5c9b1c1fcd93b97978e8b7f88db'
   BUILDOZER_SHA256SUM: '3d58a0b6972e4535718cdd6c12778170ea7382de7c75bc3728f5719437ffb84d'
-  TENSORFLOW_VERSION: 'tf-nightly==2.16.0.dev20231018'
+  TENSORFLOW_VERSION: 'tf-nightly'
+  TF_KERAS_VERSION: 'tf-keras-nightly' # Keras 2
```

Comment:
Let's also add the dependency on `tf-keras` to the requirements.txt file, since there is one case where it's used beyond tests (in hparams/_keras.py).

Reply:
Actually now that it's in the requirements.txt, I think we might need some adjustments here to the CI to avoid having both `tf-keras` and `tf-keras-nightly` installed since I would assume they probably conflict.

Perhaps the simplest would just be to skip depending on `tf-keras-nightly` at all (i.e. no special handling for it here). We install nightly TF in our CI, but I think that's mostly a special case given our close coordination with TF in the past and the need to anticipate breakages before release deadlines (plus it's easier, since TF isn't in our pip requirements). We don't depend on pre-releases for any other pip dependencies, so we could probably skip it for `tf-keras` as well.

Reply:
Ah hmmm. But maybe we can't do the simple version, if `tf-keras` itself breaks with `tf-nightly`?

Reply:
Yea, I think we have to use `tf-keras-nightly` when using `tf-nightly`: b/306638603#comment15 :(

Reply:
Ugh, ok. Maybe we can do a hacky find/replace on requirements.txt before we pip install from it then?

Reply:
Done. Thanks for the suggestion! Hope it doesn't look too hacky 😞

Reply:
Haha it looks great - sometimes a hack is the best option 😉